skynet/dgpu/__init__.py

```diff
@@ -1,41 +1,40 @@
-#!/usr/bin/python
+import logging
+from contextlib import asynccontextmanager as acm
 
 import trio
+import urwid
 
-from hypercorn.config import Config
-from hypercorn.trio import serve
-from quart_trio import QuartTrio as Quart
+from skynet.config import Config
+from skynet.dgpu.tui import init_tui
+from skynet.dgpu.daemon import dgpu_serve_forever
+from skynet.dgpu.network import NetConnector
 
-from skynet.dgpu.compute import SkynetMM
-from skynet.dgpu.daemon import SkynetDGPUDaemon
-from skynet.dgpu.network import SkynetGPUConnector
 
+@acm
+async def open_worker(config: Config):
+    # suppress logs from httpx (logs url + status after every query)
+    logging.getLogger("httpx").setLevel(logging.WARNING)
 
-async def open_dgpu_node(config: dict) -> None:
-    '''
-    Open a top level "GPU mgmt daemon", keep the
-    `SkynetDGPUDaemon._snap: dict[str, list|dict]` table
-    and *maybe* serve a `hypercorn` web API.
+    tui = None
+    if config.tui:
+        tui = init_tui(config)
 
-    '''
-    conn = SkynetGPUConnector(config)
-    mm = SkynetMM(config)
-    daemon = SkynetDGPUDaemon(mm, conn, config)
+    conn = NetConnector(config)
 
-    api: Quart|None = None
-    if 'api_bind' in config:
-        api_conf = Config()
-        api_conf.bind = [config['api_bind']]
-        api: Quart = await daemon.generate_api()
+    try:
+        n: trio.Nursery
```

Comment:
I personally prefer `tn` as short hand for `trio.Nursery` but not gonna moan about it 😂

Reply:
also, `tn` can stand for "task nursery" versus for example in `tractor` we usually suggest `an` for actor nursery.

Reply:
If we include tractor in deamon code at some point for sure lets change the names, for now dont think is necesary